### PR TITLE
StarRating docs

### PR DESCRIPTION
**Pull request by aabounegm**

### CODE REVIEW

File: docs/src/routes/docs/components/star-rating.svx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+### Functional Properties
+
+| Name | Default | Type | Description |
+| ---- | ------- | ---- | ----------- |
+| ==**`name`**==&nbsp;<span class="required">*</span> |  | `string` | The name to assign to all stars belonging to the same group. Check [MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#htmlattrdefname) for more information. |
+| **`max`** | `5` | `number` | How many stars to be displayed. |
+| **`value`** | `null` | `number` | Current amount of selected stars. |
+| **`disabled`** | `false` | `boolean` | Disables star selection. |
+| **`events`** | `[]` | `Array` | List of handlers for the [Events action](): `[{ name, handler }]`. |
+| **`...`** |  | `any` | Everything else will be passed to the underlying `<input type="radio">` element. |
```
Comment (illright) [inline docs/src/routes/docs/components/star-rating.svx:55]:
```suggestion
| **`...`** |  | `any` | Everything else will be passed to the underlying `<input type="radio">` elements. |
```

File: docs/src/routes/docs/components/star-rating.svx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+<div class="required">* Required</div>
+
+### Class Properties
+
+All `<StarRating>` components use the `.star-rating` class.
+
+| Name | Default | Type | Description |
+| ---- | ------- | ---- | ----------- |
+| **`class`** | `null` | `string` | A class string to add to the component. |
+| **`starClass`** | `null` | `string` | A class string to assign to label element containing the star icon. |
```
Comment (illright) [inline docs/src/routes/docs/components/star-rating.svx:66]:
```suggestion
| **`starClass`** | `null` | `string` | A class string to assign to the `<label>` element containing the star icon. |
```

File: docs/src/routes/docs/components/star-rating.svx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+
+## SCSS Variables {#scss-variables}
+
+| Name | Description | Default |
+| ---- | ----------- | ------- |
+| **`$main`** | Affects the hover and background colors of the component and the default values of `$hover-color` and `$selected-color`. | <ColorPreview value="#4300B0" /> |
+| **`$hover-color`** | The background color of the label when hovering on it with the mouse. | `mix(white, $main, 60%)` |
+| **`$selected-color`** | The fill color of the selected stars. | `mix(white, $main, 20%)` |
+| **`$off-state`** | The color of the (unselected) label. | <ColorPreview value="#AAAAAA" /> |
```
Comment (illright) [inline docs/src/routes/docs/components/star-rating.svx:90]: Those parentheses in the description look very odd

File: docs/src/routes/docs/components/star-rating.svx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+The icon to use as the star. Defaults to [Feather icons](https://feathericons.com/?query=star)' `Star`.
+
+
+## SCSS Variables {#scss-variables}
+
+| Name | Description | Default |
+| ---- | ----------- | ------- |
+| **`$main`** | Affects the hover and background colors of the component and the default values of `$hover-color` and `$selected-color`. | <ColorPreview value="#4300B0" /> |
+| **`$hover-color`** | The background color of the label when hovering on it with the mouse. | `mix(white, $main, 60%)` |
```
Comment (illright) [inline docs/src/routes/docs/components/star-rating.svx:88]: This one and the next are not customizable, so I see no reason to list them here. Besides, their values are derived from `$main`. Instead, I suggest to update the `$main` description to incorporate both of those descriptions in some brief way

File: docs/src/routes/docs/components/star-rating.svx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+### Functional Properties
+
+| Name | Default | Type | Description |
+| ---- | ------- | ---- | ----------- |
+| ==**`name`**==&nbsp;<span class="required">*</span> |  | `string` | The name to assign to all stars belonging to the same group. Check [MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#htmlattrdefname) for more information. |
+| **`max`** | `5` | `number` | How many stars to be displayed. |
+| **`value`** | `null` | `number` | Current amount of selected stars. |
+| **`disabled`** | `false` | `boolean` | Disables star selection. |
+| **`events`** | `[]` | `Array` | List of handlers for the [Events action](): `[{ name, handler }]`. |
```
Comment (illright) [inline docs/src/routes/docs/components/star-rating.svx:54]: We should probably mention what element are these handlers attached to

Comment (aabounegm) [reply docs/src/routes/docs/components/star-rating.svx:54]: I just copied it from the button exactly as it is (including the empty link).

**Pull request merged**